synapse/push/push_tools.py

```diff
@@ -49,7 +50,33 @@ async def get_badge_count(store: DataStore, user_id: str, group_by_room: bool) -
 async def get_context_for_event(
     storage: StorageControllers, ev: EventBase, user_id: str
 ) -> Dict[str, str]:
-    ctx = {}
+    ctx: Dict[str, str] = {}
+
+    if ev.internal_metadata.outlier:
+        # We don't have state for outliers, so we can't compute the context
+        # except for invites with `invite_room_state` set..
+        if (
+            ev.type != EventTypes.Member
+            or ev.content.get("membership") != Membership.INVITE
+        ):
+            return ctx
```

Comment:
Explaining for my own understanding:

Since this is just used to create the [push `Notification` object](https://spec.matrix.org/v1.6/push-gateway-api/#post_matrixpushv1notify), it's mostly a best effort thing since it's ok to get a notification that doesn't say what the room name is as it can just fallback to the room_alias or `room_id`, etc. Which is why it's ok that we return an empty dict without any info (`sender_display_name` and `room_name` are optional)